xdsl/dialects/stencil.py

```diff
@@ -1315,7 +1379,14 @@ def verify_(self) -> None:
             o.type.elem if isinstance(o.type, ResultType) else o.type for o in self.arg
         ]
         apply = cast(ApplyOp, self.parent_op())
-        res_types = [cast(TempType[Attribute], r.type).element_type for r in apply.res]
+        if len(apply.res) > 0:
+            res_types = [
+                cast(TempType[Attribute], r.type).element_type for r in apply.res
+            ]
+        else:
+            res_types = [
+                cast(FieldType[Attribute], o.type).element_type for o in apply.dest
+            ]
```

Comment:
I think `apply.dest` always has to be present and match, while `apply.res` only has to be there and match in unbufferized mode, no?

Reply:
To simplify a lot other parts, I went with apply having to have either res or dest, never both.